includes/class-wc-payments-account.php

```diff
@@ -230,29 +230,26 @@ private function redirect_to_onboarding_page( $error_message = null ) {
 	}
 
 	/**
-	 * Whether to do a full-page redirect to the WCPay onboarding page. It has several exceptions, to prevent
-	 * "hijacking" the multiple WC/WC-Admin onboarding flows.
-	 * This function assumes that the Stripe account hasn't been setup yet.
+	 * Checks if Stripe account is connected and redirects to the onboarding page if it is not.
 	 *
-	 * @return bool True if the user should be redirected to the WCPay onboarding page, false otherwise.
+	 * @return bool True if the redirection happened.
 	 */
-	private function should_redirect_to_onboarding() {
-		// If the user is in the WCPay settings screen and hasn't onboarded yet, always redirect.
-		if ( WC_Payment_Gateway_WCPay::is_current_page_settings() ) {
-			return true;
+	public function maybe_redirect_to_onboarding() {
+		if ( wp_doing_ajax() ) {
+			return false;
 		}
 
-		return get_option( 'wcpay_should_redirect_to_onboarding', false );
-	}
+		$is_on_settings_page           = WC_Payment_Gateway_WCPay::is_current_page_settings();
+		$should_redirect_to_onboarding = get_option( 'wcpay_should_redirect_to_onboarding', false );
 
-	/**
-	 * Checks if Stripe account is connected and redirects to the onboarding page if it is not.
-	 *
-	 * @return bool True if the account is connected properly.
-	 */
-	public function check_stripe_account_status() {
-		if ( wp_doing_ajax() ) {
-			return;
+		if (
+			// If not loading the settings page...
+			! $is_on_settings_page
+			// ...and we have redirected before.
+			&& ! $should_redirect_to_onboarding
```

Comment:
It looks like we will refresh account data only on settings page. Can it cause stale cache? Especially if we consider moving all important info to account overview page.

Reply:
> cache is invalidated by the WP transient mechanism after 2 hours

Yes, I understand that, I was a bit worried that `get_cached_account_data` won't be called on other pages. It looks like the method is used in many places, so I hope it won't lead to the problems.